qiita_db/processing_job.py

```diff
@@ -128,14 +128,43 @@ def create(cls, user, parameters):
             The user executing the job
         parameters : qiita_db.software.Parameters
             The parameters of the job being executed
+        force : bool
+            Force creation on duplicated parameters
 
         Returns
         -------
         qiita_db.processing_job.ProcessingJob
             The newly created job
+
+        Notes
+        -----
+        If force is True the job is gonna be created even if another job
+        exists with the same parameters
         """
         with qdb.sql_connection.TRN:
             command = parameters.command
+
+            # check if a job with the same parameters already exists
+            sql = """SELECT processing_job_id, email, processing_job_status
+                     FROM qiita.processing_job
+                     LEFT JOIN qiita.processing_job_status
+                        USING (processing_job_status_id)
+                     WHERE command_id = %s AND processing_job_status IN (
+                        'success', 'waiting', 'running') """
+            params = " AND ".join(
+                ["command_parameters->>'%s' = '%s'" % (k, v)
+                 for k, v in viewitems(parameters.values)
+                 if isinstance(v, Iterable) and not isinstance(v, str)
+                 for v in v])
+            sql = sql + ' AND ' if params else sql
+            qdb.sql_connection.TRN.add(sql + params, [command.id])
+            existing_jobs = qdb.sql_connection.TRN.execute_fetchindex()
+            if existing_jobs and not force:
+                raise ValueError(
+                    'Cannot create job because these jobs:\n%s' % '\n'.join(
```

Comment:
"Cannot create job because it is a duplicate of these jobs"

Reply:
👍 ... maybe worth including a bit more detail:

Cannot create job because the parameters are the same as jobs that are queued, running or already have succeeded: .....